### ISSUE TITLE
LibWeb+LibJS: `await import` fails verification in HTML::incumbent_settings_object()

### ISSUE THREAD

**Issue by awesomekling** [SerenityOS/serenity#21057]
To reproduce, you need three files:
```
test.html:
============================
<script type="module" src="main.js"></script>

main.js:
============================
const module = await import('./foo.js');
console.log(module.foo());

foo.js:
============================
export function foo() { return 1; }
```

Open test.html and you'll get the following crash. Note that you have to load the content from a web server if you want to test in other browsers.

```
VERIFICATION FAILED: !event_loop.is_backup_incumbent_settings_object_stack_empty() at Userland/Libraries/LibWeb/HTML/Scripting/Environments.cpp:349

liblagom-core.so.0(ak_verification_failed+0x79) [0x7f0d04db8579]
liblagom-web.so.0 Web::HTML::incumbent_settings_object() 0xa6) [0x7f0d0407c886]
liblagom-web.so.0(+0x5f1574) [0x7f0d03df1574]
liblagom-js.so.0(+0x31baf3) [0x7f0d0351baf3]
liblagom-js.so.0 JS::Promise::perform_then(JS::Value, JS::Value, JS::GCPtr<JS::PromiseCapability>) 0x72) [0x7f0d035e02c2]
liblagom-js.so.0 JS::CyclicModule::execute_async_module(JS::VM&) 0x47f) [0x7f0d0341dd5f]
liblagom-js.so.0 JS::CyclicModule::inner_module_evaluation(JS::VM&, AK::Vector<JS::Module*, 0ul>&, unsigned int) 0x5e9) [0x7f0d0341d569]
liblagom-js.so.0 JS::CyclicModule::evaluate(JS::VM&) 0x181) [0x7f0d0341c651]
liblagom-web.so.0 Web::HTML::JavaScriptModuleScript::run(Web::HTML::JavaScriptModuleScript::PreventErrorReporting) 0x92) [0x7f0d0408bd42]
liblagom-web.so.0 Web::HTML::HTMLScriptElement::execute_script() 0x259) [0x7f0d040107e9]
liblagom-web.so.0 Web::HTML::HTMLParser::the_end() 0x1a4) [0x7f0d04043114]
liblagom-web.so.0 Web::HTML::HTMLParser::run(AK::URL const&) 0x8b) [0x7f0d04042f1b]
liblagom-web.so.0 Web::parse_document(Web::DOM::Document&, AK::Detail::ByteBuffer<32ul> const&) 0x68) [0x7f0d03edfa58]
liblagom-web.so.0 Web::FrameLoader::resource_did_load() 0x835) [0x7f0d0413d905]
liblagom-web.so.0(+0x9413d9) [0x7f0d041413d9]
liblagom-web.so.0 Web::Resource::for_each_client(AK::Function<void (Web::ResourceClient&)>) 0x104) [0x7f0d0413fd34]
liblagom-web.so.0 Web::Resource::did_load(AK::Badge<Web::ResourceLoader>, AK::Span<unsigned char const>, AK::HashMap<AK::DeprecatedString, AK::DeprecatedString, AK::CaseInsensitiveStringTraits, AK::Traits<AK::DeprecatedString>, false> const&, AK::Optional<unsigned int>) 0xa72) [0x7f0d041409a2]

```

## Funding

- We're currently trying [Polar.sh](https://polar.sh/SerenityOS) where you can help fund the work on this issue.
- Contributor(s) receive the funding once the issue is completed & confirmed.

<a href="https://polar.sh/SerenityOS/serenity/issues/21057">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/SerenityOS/serenity/issues/21057/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/SerenityOS/serenity/issues/21057/pledge.svg">
</picture>
</a>

**Comment by awesomekling** [comment]
Possible the same crash we're seeing on https://shadow.goose.icu/

```
VERIFICATION FAILED: !event_loop.is_backup_incumbent_settings_object_stack_empty() at /home/kling/src/serenity/Userland/Libraries/LibWeb/HTML/Scripting/Environments.cpp:349
/home/kling/src/serenity/Build/lagom/lib/liblagom-core.so.0(ak_verification_failed+0xef) [0x7f7a871b93ff]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0 Web::HTML::incumbent_settings_object() 0xa6) [0x7f7a86274116]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0(+0x5f7c98) [0x7f7a85ff7c98]
/home/kling/src/serenity/Build/lagom/lib/liblagom-js.so.0(+0x314503) [0x7f7a85714503]
/home/kling/src/serenity/Build/lagom/lib/liblagom-js.so.0 JS::Promise::perform_then(JS::Value, JS::Value, JS::GCPtr<JS::PromiseCapability>) 0x72) [0x7f7a857dc1b2]
/home/kling/src/serenity/Build/lagom/lib/liblagom-js.so.0 JS::CyclicModule::execute_async_module(JS::VM&) 0x48f) [0x7f7a8561953f]
/home/kling/src/serenity/Build/lagom/lib/liblagom-js.so.0 JS::CyclicModule::inner_module_evaluation(JS::VM&, AK::Vector<JS::Module*, 0ul>&, unsigned int) 0x5e9) [0x7f7a85618d39]
/home/kling/src/serenity/Build/lagom/lib/liblagom-js.so.0 JS::CyclicModule::inner_module_evaluation(JS::VM&, AK::Vector<JS::Module*, 0ul>&, unsigned int) 0x2ee) [0x7f7a85618a3e]
/home/kling/src/serenity/Build/lagom/lib/liblagom-js.so.0 JS::CyclicModule::evaluate(JS::VM&) 0x186) [0x7f7a85617f66]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0 Web::HTML::JavaScriptModuleScript::run(Web::HTML::JavaScriptModuleScript::PreventErrorReporting) 0x92) [0x7f7a862808f2]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0 Web::HTML::HTMLScriptElement::execute_script() 0x259) [0x7f7a8620b509]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0 Web::HTML::HTMLParser::the_end() 0x1aa) [0x7f7a86243dea]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0 Web::HTML::HTMLParser::run(AK::URL const&) 0x9b) [0x7f7a86243bdb]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0 Web::parse_document(Web::DOM::Document&, AK::Detail::ByteBuffer<32ul> const&) 0x61) [0x7f7a860e1c21]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0(+0x6e5cc3) [0x7f7a860e5cc3]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0(+0x766d8e) [0x7f7a86166d8e]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0(+0x7666d7) [0x7f7a861666d7]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0 Web::HTML::Task::execute() 0x35) [0x7f7a861bcc25]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0 Web::HTML::EventLoop::process() 0x40) [0x7f7a861bb290]
/home/kling/src/serenity/Build/lagom/lib/liblagom-web.so.0(+0x9ac8a0) [0x7f7a863ac8a0]
/home/kling/src/serenity/Build/lagom/lib/liblagom-core.so.0(+0x6f109) [0x7f7a87197109]
/home/kling/src/serenity/Build/lagom/lib/liblagom-core.so.0 Core::EventReceiver::dispatch_event(Core::Event&, Core::EventReceiver*) 0x75) [0x7f7a87196685]
/home/kling/src/serenity/Build/lagom/bin/WebContent(+0xf1d1e) [0x5564d0991d1e]
/lib/x86_64-linux-gnu/libQt6Core.so.6(+0x172326) [0x7f7a86d72326]
/lib/x86_64-linux-gnu/libQt6Core.so.6 QTimer::timeout(QTimer::QPrivateSignal) 0x3e) [0x7f7a86d7678e]
/lib/x86_64-linux-gnu/libQt6Core.so.6 QObject::event(QEvent*) 0x1de) [0x7f7a86d64c7e]
/lib/x86_64-linux-gnu/libQt6Core.so.6 QCoreApplication::notifyInternal2(QObject*, QEvent*) 0xf5) [0x7f7a86d27825]
/lib/x86_64-linux-gnu/libQt6Core.so.6 QTimerInfoList::activateTimers() 0x309) [0x7f7a86e623c9]
/lib/x86_64-linux-gnu/libQt6Core.so.6(+0x317374) [0x7f7a86f17374]
/lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_dispatch+0x2ad) [0x7f7a847d649d]
/lib/x86_64-linux-gnu/libglib-2.0.so.0(+0xb6178) [0x7f7a84831178]
/lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_iteration+0x30) [0x7f7a847d51b0]
/lib/x86_64-linux-gnu/libQt6Core.so.6 QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) 0x70) [0x7f7a86f14f20]
/lib/x86_64-linux-gnu/libQt6Core.so.6 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) 0x2ba) [0x7f7a86d319aa]
/home/kling/src/serenity/Build/lagom/lib/liblagom-core.so.0 Core::EventLoop::exec() 0x23) [0x7f7a8718fb03]
/home/kling/src/serenity/Build/lagom/bin/WebContent(+0xfb88c) [0x5564d099b88c]
/home/kling/src/serenity/Build/lagom/bin/WebContent(+0xfcd22) [0x5564d099cd22]
/lib/x86_64-linux-gnu/libc.so.6(+0x23a90) [0x7f7a84a23a90]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x89) [0x7f7a84a23b49]
/home/kling/src/serenity/Build/lagom/bin/WebContent(+0x6d945) [0x5564d090d945]

```

**Comment by shannonbooth** [comment]
This was actually fixed in: 2d69a009fb451c1adf9bd6750fe06a350eb2dca7
